### PR TITLE
SAA-2198: Add endpoint to get multiple appointments

### PR DESCRIPTION
**Pull request by munslowl**
So that a user can mark attendance for multiple appointments

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/appointment/AppointmentSeries.kt [hunk 1]
```diff
@@ -112,8 +112,6 @@ data class AppointmentSeries(
       field = value
     }
 
-  fun isIndividualAppointment() = appointmentType == AppointmentType.INDIVIDUAL
-
```
Comment (munslowl) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/appointment/AppointmentSeries.kt:116]: Not used

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/ScheduledInstanceController.kt [hunk 2]
```diff
@@ -39,7 +38,6 @@ import java.time.LocalDate
 @RequestMapping("/scheduled-instances", produces = [MediaType.APPLICATION_JSON_VALUE])
 class ScheduledInstanceController(
   private val scheduledInstanceService: ScheduledInstanceService,
-  private val attendancesService: AttendancesService,
```
Comment (munslowl) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/ScheduledInstanceController.kt:42]: Not used

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/appointment/AppointmentController.kt [hunk 3]
```diff
@@ -96,6 +96,47 @@ class AppointmentController(
   fun getAppointmentDetailsById(@PathVariable("appointmentId") appointmentId: Long): AppointmentDetails =
     appointmentService.getAppointmentDetailsById(appointmentId)
 
+  @PostMapping(value = ["/details"])
```
Comment (munslowl) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/appointment/AppointmentController.kt:99]: Can't decide if `/details` could be better?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/repository/appointment/AppointmentRepository.kt [hunk 4]
```diff
@@ -5,14 +5,15 @@ import org.springframework.data.jpa.repository.Query
...
 import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.entity.appointment.Appointment
-import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.entity.appointment.AppointmentSeries
 import java.time.LocalDate
 import java.time.LocalTime
 
 @Repository
 interface AppointmentRepository : JpaRepository<Appointment, Long> {
+  @Query("from Appointment a where a.appointmentId in :ids")
+  fun findByIds(ids: List<Long>): List<Appointment>
```
Comment (dafhoustonmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/repository/appointment/AppointmentRepository.kt:14]: Could use findByAppointmentIdIn signature without the query

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/appointment/AppointmentController.kt (hunk 3 above)
Comment (dafhoustonmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/appointment/AppointmentController.kt:99]: struggling to think of a better name. It's good enough I think

**Review by dafhoustonmoj — APPROVED** [review]

**Pull request merged**